sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/CosmosCatalogITest.scala

```diff
@@ -212,6 +220,236 @@ class CosmosCatalogITest extends IntegrationSpec with CosmosClient {
+      .collect() should have size 1
+
+    val dfWithInference = spark.sql(s"SELECT * FROM testCatalog.$databaseName.$viewName")
+    val rowsArrayUnfiltered= dfWithInference.collect()
+    rowsArrayUnfiltered should have size 2
+
+    val rowsArrayWithInference = dfWithInference.where("isAlive = 'true' and type = 'mouse'").collect()
+    rowsArrayWithInference should have size 1
+
+    val rowWithInference = rowsArrayWithInference(0)
+    rowWithInference.getAs[String]("name") shouldEqual "Shrodigner's mouse"
+    rowWithInference.getAs[String]("type") shouldEqual "mouse"
+    rowWithInference.getAs[Integer]("age") shouldEqual 20
+    rowWithInference.getAs[Boolean]("isAlive") shouldEqual true
+
+    val fieldNames = rowWithInference.schema.fields.map(field => field.name)
+    fieldNames.contains(CosmosTableSchemaInferrer.SelfAttributeName) shouldBe true
+    fieldNames.contains(CosmosTableSchemaInferrer.TimestampAttributeName) shouldBe true
+    fieldNames.contains(CosmosTableSchemaInferrer.ResourceIdAttributeName) shouldBe true
+    fieldNames.contains(CosmosTableSchemaInferrer.ETagAttributeName) shouldBe true
+    fieldNames.contains(CosmosTableSchemaInferrer.AttachmentsAttributeName) shouldBe true
+  }
+
+  it can "manage Cosmos view metadata in the catalog" in {
+    val databaseName = getAutoCleanableDatabaseName()
+    val containerName = RandomStringUtils.randomAlphabetic(6).toLowerCase + System.currentTimeMillis()
+    val viewNameRaw = containerName +
+      "view" +
+      RandomStringUtils.randomAlphabetic(6).toLowerCase +
+      System.currentTimeMillis()
+    val viewNameWithSchemaInference = containerName +
+      "view" +
+      RandomStringUtils.randomAlphabetic(6).toLowerCase +
+      System.currentTimeMillis()
+
+    spark.sql(s"CREATE DATABASE testCatalog.$databaseName;")
+    spark.sql(s"CREATE TABLE testCatalog.$databaseName.$containerName using cosmos.items;")
+
+    val container = cosmosClient.getDatabase(databaseName).getContainer(containerName)
+    val containerProperties = container.read().block().getProperties
+
+    // verify default partition key path is used
+    containerProperties.getPartitionKeyDefinition.getPaths.asScala.toArray should equal(Array("/id"))
+
+    // validate throughput
+    val throughput = cosmosClient.getDatabase(databaseName).getContainer(containerName).readThroughput().block().getProperties
+    throughput.getManualThroughput shouldEqual 400
+
+    for (state <- Array(true, false)) {
+      val objectNode = Utils.getSimpleObjectMapper.createObjectNode()
+      objectNode.put("name", "Shrodigner's snake")
+      objectNode.put("type", "snake")
+      objectNode.put("age", 20)
+      objectNode.put("isAlive", state)
+      objectNode.put("id", UUID.randomUUID().toString)
+      container.createItem(objectNode).block()
+    }
+
+    spark.sql(
+      s"CREATE TABLE testCatalog.$databaseName.$viewNameRaw using cosmos.items " +
+        s"TBLPROPERTIES(isCosmosView = 'True') " +
```

Comment:
would it make sense to have tests covering the expected behaviour
if `isCosmosView=False`

Reply:
Added validation and test - also a good catch - thanks!